alternatives.md

```diff
@@ -0,0 +1,129 @@
+* [srclib](#srclib)
+* [ctags](#ctags)
+* [ANTLR](#antlr)
+* [Tree\-sitter](#tree-sitter)
+* [srcML](#srcml)
+* [SmaCC](#smacc)
+
+<!-- TODO: https://github.com/oracle/opengrok/wiki/Comparison-with-Similar-Tools -->
+
+## [Kythe](http://kythe.io/)
+
+> The best way to view Kythe is as a “hub” for connecting tools for various languages, clients and build systems. By defining language-agnostic protocols and data formats for representing, accessing and querying source code information as data, Kythe allows language analysis and indexing to be run as services.
+
+> Some tools (e.g., static analyzers) already have expressive purpose-built internal representations for code. Kythe is not meant to be a universal replacement for such IRs — instead, our goal is to provide a way for such tools to capture “interesting subsets” of an analysis for sharing with other tools.
+
+Babelfish and Kythe share a goal of defining a common representation
+for concepts from different programming languages and provide a way to query it.
+Both provide a unified data format. And both are language-independent.
+
+The main difference is that Babelfish preserves all AST nodes including
+control flow and expressions, while Kythe focuses on class hierarchy,
+dependencies, etc.
+
+Babelfish provides a unified IR across languages, which is defined
+as a non-goal of Kythe.
+
+At the same time since Babelfish only processes one file at the time, it cannot
+resolve symbolic references in the code, while the goal on Kythe project is to
+provide this information with a very high accuracy.
+
+Kythe requires to instrument language compilers and build systems (by modifying
+the sources of compiler and build system), while Babelfish uses native language
+parsers to get an AST, allowing to develop a new language driver in less time.
+
+Kythe requires a project to be built to be processed because of integration with
+compilers and build systems, while Babelfish can process only individual files.
+
+## [Language Server Protocol](https://microsoft.github.io/language-server-protocol/)
+
+> A Language Server is meant to provide the language-specific smarts and communicate with development tools over a protocol that enables inter-process communication.
+
+> The idea behind the Language Server Protocol (LSP) is to standardize the protocol for how such servers and development tools communicate. This way, a single Language Server can be re-used in multiple development tools, which in turn can support multiple languages with minimal effort.
+
+LSP defines a common protocol and RPC for queries like Go-To-Definition,
+Usages, etc. But it does not define a common representation of AST because
+the goal of the project is to enable easy access to analysis that is done
+by compilers and language SDKs.
+
+Babelfish provides a common representation for ASTs allowing to use its output
+for static analysis by other tools. Babelfish cannot resolve symbolic references,
+thus it cannot be used to query usages.
```

Comment:
"query usages" is ambiguous here, maybe "query the usages of a given identifier"?